---
layout: archive
title: Zulip Chat Archive
permalink: /stream/287929-mathlib4/topic/strange.20error.20in.20.60to_additive.60.html
---

<h2>Stream: <a href="https://leanprover-community.github.io/archive/stream/287929-mathlib4/index.html">mathlib4</a></h2>
<h3>Topic: <a href="https://leanprover-community.github.io/archive/stream/287929-mathlib4/topic/strange.20error.20in.20.60to_additive.60.html">strange error in `to_additive`</a></h3>

<hr>

<base href="https://leanprover.zulipchat.com/">

<head><link href="/style.css" rel="stylesheet"></head>

{% raw %}

<a name="319297994"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/strange%20error%20in%20%60to_additive%60/near/319297994" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Arien Malec <a href="https://leanprover-community.github.io/archive/stream/287929-mathlib4/topic/strange.20error.20in.20.60to_additive.60.html#319297994">(Jan 03 2023 at 23:26)</a>:</h4>
<p>I'm porting <code>Algebra.PUnitInstances</code> (<a href="https://github.com/leanprover-community/mathlib4/pull/1319">mathlib4#1319</a>) and getting very odd errors with <code>to_additive</code> like</p>
<div class="codehilite" data-code-language="Lean"><pre><span></span><code><span class="n">application</span> <span class="n">type</span> <span class="n">mismatch</span>
  <span class="n">HSMul.hSMul</span>
<span class="n">argument</span> <span class="n">has</span> <span class="n">type</span>
  <span class="n">HVAdd</span> <span class="n">R</span> <span class="n">PUnit</span> <span class="n">PUnit</span>
<span class="n">but</span> <span class="n">function</span> <span class="n">has</span> <span class="n">type</span>
  <span class="o">[</span><span class="n">self</span> <span class="o">:</span> <span class="n">HSMul</span> <span class="n">R</span> <span class="n">PUnit</span> <span class="n">PUnit</span><span class="o">]</span> <span class="bp">→</span> <span class="n">R</span> <span class="bp">→</span> <span class="n">PUnit</span> <span class="bp">→</span> <span class="n">PUnit</span>
</code></pre></div>
<p>as if <code>to_additive</code> is getting confused?</p>



<a name="319298130"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/strange%20error%20in%20%60to_additive%60/near/319298130" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Riccardo Brasca <a href="https://leanprover-community.github.io/archive/stream/287929-mathlib4/topic/strange.20error.20in.20.60to_additive.60.html#319298130">(Jan 03 2023 at 23:27)</a>:</h4>
<p>This also happens in <a href="https://github.com/leanprover-community/mathlib4/pull/1238">mathlib4#1238</a></p>



<a name="319385061"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/strange%20error%20in%20%60to_additive%60/near/319385061" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Floris van Doorn <a href="https://leanprover-community.github.io/archive/stream/287929-mathlib4/topic/strange.20error.20in.20.60to_additive.60.html#319385061">(Jan 04 2023 at 13:04)</a>:</h4>
<p><span class="user-mention" data-user-id="542196">@Arien Malec</span> there was some configuration of <code>to_additive</code> that was not ported over from Lean 3. Can you merge your branch with <a href="https://github.com/leanprover-community/mathlib4/pull/1330">mathlib4#1330</a>, and see if that fixes it?</p>



<a name="319395353"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/strange%20error%20in%20%60to_additive%60/near/319395353" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Floris van Doorn <a href="https://leanprover-community.github.io/archive/stream/287929-mathlib4/topic/strange.20error.20in.20.60to_additive.60.html#319395353">(Jan 04 2023 at 14:04)</a>:</h4>
<p><a href="https://github.com/leanprover-community/mathlib4/pull/1332">mathlib4#1332</a> fixes the issue in <a href="https://github.com/leanprover-community/mathlib4/pull/1238">mathlib4#1238</a></p>



<a name="319436004"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/strange%20error%20in%20%60to_additive%60/near/319436004" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Arien Malec <a href="https://leanprover-community.github.io/archive/stream/287929-mathlib4/topic/strange.20error.20in.20.60to_additive.60.html#319436004">(Jan 04 2023 at 17:29)</a>:</h4>
<p>Confirmed fixed.</p>



{% endraw %}

<hr><p>Last updated: Jan 25 2023 at 00:06 UTC</p>